sprocs/jsonb_array_to_double_precision_array.sql

```diff
@@ -0,0 +1,27 @@
+DROP FUNCTION IF EXISTS jsonb_array_to_double_precision_array(jsonb, double precision[]);
+
+CREATE OR REPLACE FUNCTION
+    jsonb_array_to_double_precision_array(
+        IN jsonb_array jsonb,
+        OUT double_precision_array double precision[]
+    )
+AS $$
+DECLARE
+    len integer;
+BEGIN
+    IF jsonb_array = 'null'::jsonb THEN
+        double_precision_array := NULL;
+        RETURN;
+    END IF;
+
+    len := jsonb_array_length(jsonb_array);
+
+    IF len IS NULL THEN
+        double_precision_array := NULL;
+    ELSIF len = 0 THEN
+        double_precision_array := array[]::double precision[];
+    ELSE
+        double_precision_array := (SELECT array_agg(v::double precision) FROM jsonb_array_elements_text(jsonb_array) AS tmp(v));
```

Comment:
The reason for `AS tmp(v)` is not obvious to me here.

Reply:
It's a table alias, so I don't have to write
```
array_agg(jsonb_array_elements_text::double precision)
```

The general table alias syntax is `AS tablename(col1name, col2name)`. You don't have to provide the column aliases if you don't want them. In this case they aren't strictly necessary, so I just pushed a change `tmp(v) -> v`.